lib/http3/server.c

```diff
@@ -984,6 +984,8 @@ static int handle_input_expect_headers(struct st_h2o_http3_server_stream_t *stre
         return 0;
     }
 
+    h2o_probe_log_request(&stream->req, stream->quic->stream_id);
```

Comment:
Should we better log the request headers at an earlier point? I'd assume that we'd want to log the request headers whenever we log the response headers. However, this line would never be executed when `h2o_send_error_4xx` functions are called in the lines above.

Reply:
Yes, indeed. Thank you for pointing out my shortsightedness. I tweaked the code so that the `h2o_probe_log_request` is called on error as well.